src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.ProjectStates.cs

```diff
@@ -97,81 +97,21 @@ private ProjectAnalyzerStateSets CreateProjectStateSets(Project project)
             /// </summary>
             private async Task<ProjectAnalyzerStateSets> UpdateProjectStateSetsAsync(Project project, CancellationToken cancellationToken)
             {
-                ProjectAnalyzerReferenceChangedEventArgs? analyzerReferenceChangedEventArgs = null;
-                ProjectAnalyzerStateSets? projectStateSets;
-
                 // This code is called concurrently for a project, so the guard prevents duplicated effort calculating StateSets.
                 using (await _projectAnalyzerStateMapGuard.DisposableWaitAsync(cancellationToken).ConfigureAwait(false))
                 {
-                    projectStateSets = TryGetProjectStateSets(project);
+                    var projectStateSets = TryGetProjectStateSets(project);
 
                     if (projectStateSets == null)
                     {
                         projectStateSets = CreateProjectStateSets(project);
-                        analyzerReferenceChangedEventArgs = GetProjectAnalyzerReferenceChangedEventArgs(project);
 
                         // update cache. 
                         _projectAnalyzerStateMap = _projectAnalyzerStateMap.SetItem(project.Id, projectStateSets.Value);
                     }
-                }
-
-                if (analyzerReferenceChangedEventArgs != null)
-                    RaiseProjectAnalyzerReferenceChanged(analyzerReferenceChangedEventArgs);
```

Comment:
this event existed to have us drop cached data for a particular project.  but as data is cached with a CWT against an actual Project instance, this comes for free as changing the project analyzer references will cause a new instance ot be created.